src/sources/vector/v2.rs

```diff
@@ -28,20 +29,32 @@ impl proto::Service for Service {
         &self,
         request: Request<proto::EventRequest>,
     ) -> Result<Response<proto::EventResponse>, Status> {
-        let event = request
+        let (event, receiver) = request
             .into_inner()
             .message
-            .map(Event::from)
+            .map(|data| {
+                let (batch, receiver) = BatchNotifier::new_with_receiver();
+                let event = Event::from(data).with_batch_notifier(&batch);
+                (event, receiver)
+            })
             .ok_or_else(|| Status::invalid_argument("missing event"))?;
 
-        let response = Response::new(proto::EventResponse {});
-
         self.pipeline
             .clone()
             .send(event)
             .await
-            .map(|_| response)
-            .map_err(|err| Status::unavailable(err.to_string()))
+            .map_err(|err| Status::unavailable(err.to_string()))?;
+
+        let status = if self.acknowledgements {
+            receiver.await.unwrap_or(BatchStatus::Delivered)
```

Comment:
Is this correct? If I read this correctly, if the receiver channel drops without sending, we are saying the Batch was delivered.

Reply:
This is the precedent we set in the sources using `util::http`, but that is worth reviewing too.
https://github.com/timberio/vector/blob/19fb08ab995991cdbc666a287cbee67a144fbbc9/src/sources/util/http.rs#L328
The `Delivered` default was there to prevent a behavior change, but reporting `Errored` would probably make more sense in retrospect. We should also be making some noise about this situation, as it's an internal programming error not some other expected condition, so a wrapper function would probably be good. I'll open a separate PR for that.